# Load the packages

In [1]:
import os
import re
import pandas as pd
import matplotlib.pyplot as plt
import seaborn as sns

%matplotlib inline

# Load the data 

In [2]:
train = pd.read_csv("Data/Train.csv", low_memory=False)
test = pd.read_csv("Data/Test.csv", low_memory=False)
sample_submission = pd.read_csv('Data/SampleSubmission.csv') 

In [3]:
train.head()

child_id  data_year  child_date  child_age child_enrolment_date  \
0  ID_SYSJ2FM0D     2022.0  2022-02-03  59.000000                  NaN   
1  ID_J5BTFOZR3     2019.0         NaN  60.163933                  NaN   
2  ID_R00SN7AUD     2022.0  2022-03-11  69.000000                  NaN   
3  ID_BSSK60PAZ     2021.0  2021-10-13  53.000000           2020-01-15   
4  ID_IZTY6TC4D     2021.0  2021-10-13  57.000000           2021-10-13   

   child_months_enrolment child_grant   child_years_in_programme  \
0                     NaN         NaN                        NaN   
1                     NaN         NaN  1st year in the programme   
2                     NaN         NaN                        NaN   
3                    20.0          No  1st year in the programme   
4                     0.0         NaN      2nd year in programme   

   child_height child_observe_attentive  ... obs_cooking_5 obs_cooking_6  \
0           NaN               Sometimes  ...           NaN           NaN   
1    103.000000               Sometimes  ...           NaN           NaN   
2    108.400002                   Often  ...           NaN           NaN   
3     98.099998           Almost always  ...           NaN           NaN   
4    114.000000           Almost always  ...           0.0           0.0   

  obs_heating_1  obs_heating_2 obs_heating_3 obs_heating_4  obs_heating_5  \
0           NaN            NaN           NaN           NaN            NaN   
1           NaN            NaN           NaN           NaN            NaN   
2           NaN            NaN           NaN           NaN            NaN   
3           NaN            NaN           NaN           NaN            NaN   
4           1.0            0.0           0.0           0.0            0.0   

  obs_heating_6 obs_heating_7     target  
0           NaN           NaN  51.500000  
1           NaN           NaN  55.869999  
2           NaN           NaN  47.520000  
3           NaN           NaN  58.599998  
4           0.0           0.0  76.599998  

[5 rows x 679 columns]

In [4]:
train.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 8585 entries, 0 to 8584
Columns: 679 entries, child_id to target
dtypes: float64(160), object(519)
memory usage: 44.5+ MB


In [5]:
test.head()

child_id  data_year  child_date  child_age child_enrolment_date  \
0  ID_0I0999N6S     2021.0  2021-09-20  57.000000                  NaN   
1  ID_GQ6ONJ4FP     2021.0  2021-10-21  54.000000           2021-01-10   
2  ID_YZ76CVRW3     2021.0  2021-05-17  57.000000                  NaN   
3  ID_BNINCRXH8     2022.0  2022-09-09  59.334702                  NaN   
4  ID_1U7GDTLRI     2021.0  2021-10-12  54.000000           2021-01-15   

   child_months_enrolment child_grant   child_years_in_programme  \
0                     NaN         Yes      2nd year in programme   
1                     9.0         Yes  1st year in the programme   
2                     NaN         Yes                        NaN   
3                     NaN         NaN      3rd year in programme   
4                     8.0         Yes  1st year in the programme   

   child_height child_observe_attentive  ... obs_cooking_4 obs_cooking_5  \
0         108.0           Almost always  ...           NaN           NaN   
1         105.0           Almost always  ...           NaN           NaN   
2         101.5                   Often  ...           NaN           NaN   
3           NaN           Almost always  ...           0.0           0.0   
4         103.5                   Often  ...           NaN           NaN   

  obs_cooking_6  obs_heating_1 obs_heating_2 obs_heating_3  obs_heating_4  \
0           NaN            NaN           NaN           NaN            NaN   
1           NaN            NaN           NaN           NaN            NaN   
2           NaN            NaN           NaN           NaN            NaN   
3           0.0            0.0           0.0           0.0            1.0   
4           NaN            NaN           NaN           NaN            NaN   

  obs_heating_5 obs_heating_6  obs_heating_7  
0           NaN           NaN            NaN  
1           NaN           NaN            NaN  
2           NaN           NaN            NaN  
3           0.0           0.0            0.0  
4           NaN           NaN            NaN  

[5 rows x 678 columns]

In [6]:
test.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 3680 entries, 0 to 3679
Columns: 678 entries, child_id to obs_heating_7
dtypes: float64(161), object(517)
memory usage: 19.0+ MB


In [7]:
sample_submission.head()

child_id  target feature_1 feature_2 feature_3 feature_4 feature_5  \
0  ID_0I0999N6S       0   feature   feature   feature   feature   feature   
1  ID_GQ6ONJ4FP       0   feature   feature   feature   feature   feature   
2  ID_YZ76CVRW3       0   feature   feature   feature   feature   feature   
3  ID_BNINCRXH8       0   feature   feature   feature   feature   feature   
4  ID_1U7GDTLRI       0   feature   feature   feature   feature   feature   

  feature_6 feature_7 feature_8 feature_9 feature_10 feature_11 feature_12  \
0   feature   feature   feature   feature    feature    feature    feature   
1   feature   feature   feature   feature    feature    feature    feature   
2   feature   feature   feature   feature    feature    feature    feature   
3   feature   feature   feature   feature    feature    feature    feature   
4   feature   feature   feature   feature    feature    feature    feature   

  feature_13 feature_14 feature_15  
0    feature    feature    feature  
1    feature    feature    feature  
2    feature    feature    feature  
3    feature    feature    feature  
4    feature    feature    feature

In [8]:
sample_submission.head()

child_id  target feature_1 feature_2 feature_3 feature_4 feature_5  \
0  ID_0I0999N6S       0   feature   feature   feature   feature   feature   
1  ID_GQ6ONJ4FP       0   feature   feature   feature   feature   feature   
2  ID_YZ76CVRW3       0   feature   feature   feature   feature   feature   
3  ID_BNINCRXH8       0   feature   feature   feature   feature   feature   
4  ID_1U7GDTLRI       0   feature   feature   feature   feature   feature   

  feature_6 feature_7 feature_8 feature_9 feature_10 feature_11 feature_12  \
0   feature   feature   feature   feature    feature    feature    feature   
1   feature   feature   feature   feature    feature    feature    feature   
2   feature   feature   feature   feature    feature    feature    feature   
3   feature   feature   feature   feature    feature    feature    feature   
4   feature   feature   feature   feature    feature    feature    feature   

  feature_13 feature_14 feature_15  
0    feature    feature    feature  
1    feature    feature    feature  
2    feature    feature    feature  
3    feature    feature    feature  
4    feature    feature    feature

In [9]:
sample_submission.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 3680 entries, 0 to 3679
Data columns (total 17 columns):
 #   Column      Non-Null Count  Dtype 
---  ------      --------------  ----- 
 0   child_id    3680 non-null   object
 1   target      3680 non-null   int64 
 2   feature_1   3680 non-null   object
 3   feature_2   3680 non-null   object
 4   feature_3   3680 non-null   object
 5   feature_4   3680 non-null   object
 6   feature_5   3680 non-null   object
 7   feature_6   3680 non-null   object
 8   feature_7   3680 non-null   object
 9   feature_8   3680 non-null   object
 10  feature_9   3680 non-null   object
 11  feature_10  3680 non-null   object
 12  feature_11  3680 non-null   object
 13  feature_12  3680 non-null   object
 14  feature_13  3680 non-null   object
 15  feature_14  3680 non-null   object
 16  feature_15  3680 non-null   object
dtypes: int64(1), object(16)
memory usage: 488.9+ KB


In [10]:
variables_description = pd.read_csv("Data/VariableDescription.csv")
variables_description.head()

Variable Name           Variable Label Answer Label
0              child_id          Unique child ID   Open ended
1             data_year  Year data was collected   Open ended
2            child_date                ELOM date   Open ended
3             child_age      Child age in months   Open ended
4  child_enrolment_date     Date enrolled in ELP   Open ended

For the above cells we learned that the train set has 8585 observations 679 columns with the data types: 160 are float64, 519 are object.

Let's first make sure the data type of each column is the appropriate one.

In [21]:
# we select data types that are float first

train_number_df = train.select_dtypes('number').copy()

In [38]:
train_number_df

data_year  child_age  child_months_enrolment  child_height  \
0        2022.0  59.000000                     NaN           NaN   
1        2019.0  60.163933                     NaN    103.000000   
2        2022.0  69.000000                     NaN    108.400002   
3        2021.0  53.000000                    20.0     98.099998   
4        2021.0  57.000000                     0.0    114.000000   
...         ...        ...                     ...           ...   
8580     2021.0  55.000000                     9.0    102.300003   
8581     2021.0  55.000000                    32.0    102.599998   
8582     2021.0  56.000000                    45.0    103.800003   
8583     2021.0  57.000000                     9.0    102.400002   
8584     2021.0  50.000000                     NaN    101.000000   

      child_observe_total  child_zha  child_attendance  ward_best  \
0                     4.0        NaN               NaN       39.0   
1                     4.0  -1.356791               NaN        NaN   
2                     7.0  -1.250863               NaN       85.0   
3                     9.0  -1.830364               NaN       18.0   
4                    12.0   1.329264               NaN       10.0   
...                   ...        ...               ...        ...   
8580                  6.0  -0.968543               NaN        2.0   
8581                  4.0  -1.037532               NaN       11.0   
8582                 12.0  -0.884394               NaN       25.0   
8583                 12.0  -1.166885               NaN       47.0   
8584                 12.0  -0.661874               NaN       48.0   

      id_enumerator  id_facility  ...  obs_cooking_5  obs_cooking_6  \
0           20005.0        761.0  ...            NaN            NaN   
1               NaN        458.0  ...            NaN            NaN   
2           20001.0        925.0  ...            NaN            NaN   
3            2689.0        308.0  ...            NaN            NaN   
4             542.0       1749.0  ...            0.0            0.0   
...             ...          ...  ...            ...            ...   
8580         2795.0       1371.0  ...            0.0            0.0   
8581          567.0       1741.0  ...            0.0            0.0   
8582         2793.0       1490.0  ...            0.0            0.0   
8583         2985.0       1718.0  ...            NaN            NaN   
8584         2985.0        253.0  ...            NaN            NaN   

      obs_heating_1  obs_heating_2  obs_heating_3  obs_heating_4  \
0               NaN            NaN            NaN            NaN   
1               NaN            NaN            NaN            NaN   
2               NaN            NaN            NaN            NaN   
3               NaN            NaN            NaN            NaN   
4               1.0            0.0            0.0            0.0   
...             ...            ...            ...            ...   
8580            0.0            1.0            0.0            0.0   
8581            0.0            1.0            0.0            0.0   
8582            0.0            1.0            0.0            0.0   
8583            NaN            NaN            NaN            NaN   
8584            NaN            NaN            NaN            NaN   

      obs_heating_5  obs_heating_6  obs_heating_7     target  
0               NaN            NaN            NaN  51.500000  
1               NaN            NaN            NaN  55.869999  
2               NaN            NaN            NaN  47.520000  
3               NaN            NaN            NaN  58.599998  
4               0.0            0.0            0.0  76.599998  
...             ...            ...            ...        ...  
8580            0.0            0.0            0.0  23.469999  
8581            0.0            0.0            0.0  45.639999  
8582            0.0            0.0            0.0  53.290001  
8583            NaN            NaN            NaN  56.330002

data_year:  int
child_age: ??
child_months_enrolment : ??
child_attendance : int
war_best : ??

In [25]:
# view the full list of columns and check their meaning from the VariableDescription.csv

list(train_number_df.columns)

['data_year',
 'child_age',
 'child_months_enrolment',
 'child_height',
 'child_observe_total',
 'child_zha',
 'child_attendance',
 'ward_best',
 'id_enumerator',
 'id_facility',
 'pqa_class_assistants',
 'pqa_score_environment',
 'pqa_score_assessment',
 'pqa_score_relationships',
 'pqa_score_curriculum',
 'pqa_score_teaching',
 'pra_experience',
 'pra_class_present',
 'pra_class_size',
 'pra_class_attendance',
 'pra_class_attendance_precovid',
 'pra_hhsize',
 'pra_measure_rectangle_length',
 'pra_measure_rectangle_width',
 'pra_class_space',
 'pri_days',
 'pri_time_open_hours',
 'pri_time_open_minutes',
 'pri_time_close_hours',
 'pri_time_close_minutes',
 'pri_year',
 'pri_fees_amount',
 'pri_dsd_year',
 'pri_capacity',
 'pri_amount_funding_dsd',
 'pri_amount_funding_fees',
 'pri_children_4_6_years',
 'pri_difficult_see',
 'pri_difficult_hear',
 'pri_difficult_walk',
 'pri_difficult_hold',
 'pri_difficult_communicate',
 'pri_difficult_learn',
 'pri_covid_staff_retrench',
 'pri_fees_a